localstack/services/awslambda/invocation/runtime_executor_plugin.py

```diff
@@ -0,0 +1,113 @@
+import logging
+from abc import ABC, abstractmethod
+from typing import Type
+
+from plugin import Plugin, PluginManager
+
+from localstack import config
+from localstack.aws.api.lambda_ import FunctionVersion
+from localstack.services.awslambda.invocation.lambda_models import ServiceEndpoint
+
+LOG = logging.getLogger(__name__)
+
+
+class RuntimeExecutorPlugin(Plugin):
+    namespace = "localstack.lambda.runtime_executor"
+
+
+class RuntimeExecutor(ABC):
+    id: str
+    function_version: FunctionVersion
+
+    def __init__(
+        self, id: str, function_version: FunctionVersion, service_endpoint: ServiceEndpoint
+    ) -> None:
+        """
+        Runtime executor class responsible for executing a runtime in specific environment
+
+        :param id: ID string of the runtime executor
+        :param function_version: Function version to be executed
+        :param service_endpoint: Service endpoint for execution related callbacks
+        """
+        self.id = id
+        self.function_version = function_version
+
+    @abstractmethod
+    def start(self, env_vars: dict[str, str]) -> None:
+        """
+        Start the runtime executor with the given environment variables
+
+        :param env_vars:
+        """
+        pass
+
+    @abstractmethod
+    def stop(self) -> None:
+        """
+        Stop the runtime executor
+        """
+        pass
+
+    @abstractmethod
+    def get_address(self) -> str:
+        """
+        Get the address the runtime executor is available at for the LocalStack container.
+
+        :return: IP address or hostname of the execution environment
+        """
+        pass
+
+    @abstractmethod
+    def get_endpoint_from_executor(self) -> str:
+        """
+        Get the address of LocalStack the runtime execution environment can communicate with LocalStack
+
+        :return: IP address or hostname of LocalStack (from the view of the execution environment)
+        """
+        pass
+
+    @abstractmethod
+    def invoke(self, payload: dict[str, str]) -> None:
+        """
+        Send an invocation to the execution environment
+
+        :param payload: Invocation payload
+        """
+        pass
+
+    @staticmethod
+    @abstractmethod
+    def prepare_version(function_version: FunctionVersion) -> None:
```

Comment:
nit: static abstract methods don't make a whole lot of sense IMO. unless there is some specific reason why they have to be static/class methods, i would just make normal methods out of them, or at least classmethods

Reply:
Can make them classmethods, normal methods do not make too much sense. I want to encapsulate the prepare/cleanup version routines with the executor implementation (as those two are completely connected) while also being able to execute it before instantiating the first executor.